### ISSUE TITLE
optomized code to obtain headerNumber when `position=0` and fix unwanted  java.io.EOFException

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Queue#495]
fixed

```
15:30:28.380 [main/writer-1] DEBUG net.openhft.chronicle.queue.impl.single.SCQIndexing - Attempt to find 0
java.io.EOFException
	at net.openhft.chronicle.wire.AbstractWire.writeHeader0(AbstractWire.java:457)
	at net.openhft.chronicle.wire.AbstractWire.writeHeaderOfUnknownLength(AbstractWire.java:299)
	at net.openhft.chronicle.queue.impl.single.StoreRecovery.writeHeader(StoreRecovery.java:50)
	at net.openhft.chronicle.queue.impl.single.SCQIndexing.newIndex(SCQIndexing.java:252)
	at net.openhft.chronicle.queue.impl.single.SCQIndexing.newIndex(SCQIndexing.java:268)
	at net.openhft.chronicle.queue.impl.single.SCQIndexing.getSecondaryAddress1(SCQIndexing.java:608)
	at net.openhft.chronicle.queue.impl.single.SCQIndexing.getSecondaryAddress(SCQIndexing.java:594)
	at net.openhft.chronicle.queue.impl.single.SCQIndexing.sequenceForPosition(SCQIndexing.java:521)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueStore.sequenceForPosition(SingleChronicleQueueStore.java:364)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts$StoreAppender.resetPosition(SingleChronicleQueueExcerpts.java:371)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts$StoreAppender.setCycle2(SingleChronicleQueueExcerpts.java:322)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts$StoreAppender.writingDocument(SingleChronicleQueueExcerpts.java:398)
	at net.openhft.chronicle.wire.MarshallableOut.writingDocument(MarshallableOut.java:57)
	at net.openhft.chronicle.queue.impl.single.RollCycleMultiThreadStressTest$Writer.write(RollCycleMultiThreadStressTest.java:344)
	at net.openhft.chronicle.queue.impl.single.RollCycleMultiThreadStressTest$Writer.call(RollCycleMultiThreadStressTest.java:321)
	at net.openhft.chronicle.queue.impl.single.RollCycleMultiThreadStressTest$Writer.call(RollCycleMultiThreadStressTest.java:299)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by hft-team-city** [comment]
Released in Chronicle-Queue-4.16.4 (https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-4.16.4), BOM-1.16.32 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-1.16.32)

**Comment by hft-team-city** [comment]
Released in Chronicle-Queue-4.16.4 (https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-4.16.4), BOM-1.16.39 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-1.16.39)
